src/librustc_mir/borrow_check/nll/liveness_map.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use rustc_data_structures::indexed_vec::IndexVec;
+use rustc::mir::{Mir, Local};
+use util::liveness::LiveVariableMap;
+use rustc_data_structures::indexed_vec::Idx;
+use rustc::ty::TypeFoldable;
+
+crate struct NllLivenessMap {
+    pub from_local: IndexVec<Local, Option<LocalWithRegion>>,
+    pub to_local: IndexVec<LocalWithRegion, Local>,
```

Comment:
Comment:

---

For each `LocalWithRegion`, maps back to the original `Local` index.